genetic_algorithms/solvers/solver.py

```diff
@@ -17,7 +18,8 @@ class SolverConfig:
 
 class Solver(ABC):
 
-    def __init__(self, config: SolverConfig = None):
+    def __init__(self, algorithm: Algorithm, config: SolverConfig = None):
+        self.algorithm = algorithm
```

Comment:
Ja bym nie dawał algorytmu tutaj, a tylko zrobiłbym tak, że my będziemy przekazywać algorytm do metody `solve` solvera.
Myślę, że to by było sensowne dlatego, że wtedy moglibyśmy rozwiązywać nasze problemy nie tworząc za każdym razem nowej instancji solver'a. 


Ale z innej strony, tworzenie `solvera` bez przypisanego algorytmu jest dziwne. 
Co ty o tym myślisz?

Reply:
Jakoś mi się wydało naturalniejsze przypisać algorytm do solvera, ale myślę, że faktycznie więcej sensu ma przekazywanie go do `solve`